service/src/main/java/org/cbioportal/service/impl/CoExpressionServiceImpl.java

```diff
@@ -42,65 +55,171 @@
     private GenesetDataService genesetDataService;
     @Autowired
     private MolecularProfileService molecularProfileService;
-    
+    @Autowired
+    private SampleListRepository sampleListRepository;
+    @Autowired
+    private MolecularDataRepository molecularDataRepository;
+    @Autowired
+    private SampleService sampleService;
+
+    @Override
+    public List<CoExpression> getCoExpressions(String molecularProfileId, String sampleListId, String geneticEntityId,
+            CoExpression.GeneticEntityType geneticEntityType, Double threshold)
+            throws MolecularProfileNotFoundException, GenesetNotFoundException, GeneNotFoundException {
+
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        return fetchCoExpressions(molecularProfileId, sampleIds, geneticEntityId, geneticEntityType, threshold);
+    }
+
     @Override
-    public List<CoExpression> getCoExpressions(String geneticEntityId, CoExpression.GeneticEntityType geneticEntityType, String sampleListId, 
-                                              String molecularProfileIdA, String molecularProfileIdB, Double threshold) throws Exception {
+    public List<CoExpression> getCoExpressions(String geneticEntityId, CoExpression.GeneticEntityType geneticEntityType,
```

Comment:
this is called overloading yes? same method, two different signatures?  so this one handles different profileIds, yes?

Reply:
yes